src/library/parser.cpp

```diff
@@ -158,3 +170,50 @@ TrackFile Parser::playlistEntryToTrackFile(
         return TrackFile(QDir(basePath), filePath);
     }
 }
+
+bool Parser::exportPlaylistItemsIntoFile(
+        QString playlistFilePath,
+        const QList<QString>& playlistItemLocations,
+        bool useRelativePath) {
+    if (playlistFilePath.endsWith(
+                QStringLiteral(".pls"),
+                Qt::CaseInsensitive)) {
+        return ParserPls::writePLSFile(
+                playlistFilePath,
+                playlistItemLocations,
+                useRelativePath);
+    } else if (playlistFilePath.endsWith(
+                       QStringLiteral(".m3u8"),
+                       Qt::CaseInsensitive)) {
+        return ParserM3u::writeM3U8File(
+                playlistFilePath,
+                playlistItemLocations,
+                useRelativePath);
+    } else {
+        //default export to M3U if file extension is missing
+        if (!playlistFilePath.endsWith(
+                    QStringLiteral(".m3u"),
+                    Qt::CaseInsensitive)) {
+            kLogger.debug()
+                    << "No valid file extension for playlist export specified."
+                    << "Appending .m3u and exporting to M3U.";
+            playlistFilePath.append(QStringLiteral(".m3u"));
+            if (QFileInfo::exists(playlistFilePath)) {
+                auto overwrite = QMessageBox::question(
+                        nullptr,
+                        tr("Overwrite File?"),
```

Comment:
This sound techy to me. How about 

How about: 
"Playlist file ready exists"

"A playlist file with the name \"%1\" already exists.\n"
"The default \"m3u\" extension was added because none was specified.\n\n"
"Do you really want to replace it?"

Reply:
Why do we have only the message box in case the extension was added?

Reply:
Jup, let's have a common way to deal with file overwrite.
For replacing an exitisting controller mapping there is:
https://github.com/mixxxdj/mixxx/blob/main/src/controllers/dlgprefcontroller.cpp#L559-L563
```
        QString overwriteTitle = tr("Mapping already exists.");
        QString overwriteLabel = tr(
                "<b>%1</b> already exists in user mapping folder.<br>"
                "Overwrite or save with a new name?");
        QString overwriteCheckLabel = tr("Always overwrite during this session");
```